cmd/krew/cmd/install.go

```diff
@@ -97,6 +98,10 @@ Remarks:
 
 			var install []pluginEntry
 			for _, name := range pluginNames {
+				if !validation.IsSafePluginName(name) {
+					return unsafePluginNameErr(name)
+				}
```

Comment:
I'd feel better if we validate plugin names in batch upfront. One way we could achieve this is by changing the API to
```go
func IsSafePluginName(names ...string) bool { .. }
```

(same in several other places)

Also, it seems that `unsafePluginNameErr()` could be returned from this function (making it obsolete), so that we have instead:
```go
func CheckSafePluginName(names ...string) error { .. }
```

Reply:
yeah although there's another method in `validation.go` that's geared towards plugin authors (and therefore it prints a regexp) as it runs during manifest validation tests.

    return errors.Errorf("the plugin name %q is not allowed, must match %q", name, safePluginRegexp.String())

that's why I kept it like this.

Reply:
I understand that you want to keep the API in `validation` consistent.

Maybe this helper can be put in `cmd/internal` instead? Just an idea. It's fine as it stands, but validating upfront would be even better IMO.